docs/topics/extraction.md

```diff
@@ -1,5 +1,115 @@
 # Extraction of Salt core modules
-TODO - ref https://github.com/salt-extensions/extension_migration
+
+Below are some rough steps that extract an existing set of modules into an extension while preserving the Git history. Let's use the `stalekey` engine as an example.
+
+## A module extraction example
+
+### 1. Install the Git history filtering tool
+
+```shell
+pip install git-filter-repo
+```
+
+### 2. Clone the Salt repo and analyze its history
+
+```shell
+git clone https://github.com/saltstack/salt --single-branch
+cd salt
+git filter-repo --analyze
+tree .git/filter-repo/analysis/
+rg stalekey .git/filter-repo/analysis/path-all-sizes.txt
```

Comment:
question: Is ripgrep required or does grep work just as fine? If the former, I think we should a Prerequisites heading and tell people to install it.

Reply:
Grep will work just fine, ripgrep is just what I had in my notes.

Reply:
Let's `%s/rg/grep/g` then for minimalism's sake :)